packages/muelu/src/MueCentral/MueLu_VerbosityLevel.hpp

```diff
@@ -109,6 +110,40 @@ namespace MueLu {
   //!
   VerbLevel toMueLuVerbLevel(const Teuchos::EVerbosityLevel verbLevel);
 
+  //! global variable that holds the default output rank for new VerboseObjects
+  extern int defaultMueLuOutputRank;
+
+  //! constant that is interpreted as "output on default rank"
+  extern const int OUTPUT_TO_ROOT;
+
+  //! On all MPI ranks that are contained in oldMap, set the default output rank for MueLu to the lowest available rank that is present in newMap.
+  //! This is used in rebalancing, since the original root output might have been dropped.
+  template <class LocalOrdinal, class GlobalOrdinal, class Node>
+  void updateDefaultMueLuOutputRank(Teuchos::RCP<const typename Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> > oldMap,
+                                    Teuchos::RCP<const typename Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> > newMap) {
+#ifdef HAVE_MPI
+    Teuchos::RCP<Teuchos::Comm<int> > oldComm = Teuchos::rcp_const_cast<Teuchos::Comm<int> >(oldMap->getComm());
+    if (newMap == Teuchos::null || oldComm->getSize() != newMap->getComm()->getSize()) {
+      int mpiStarted = 0;
+      MPI_Initialized(&mpiStarted);
+      if (mpiStarted) {
+        int globalRank = 0;
+        int globalSize = 0;
+        MPI_Comm_rank(MPI_COMM_WORLD, &globalRank);
+        MPI_Comm_size(MPI_COMM_WORLD, &globalSize);
+
+        if (newMap == Teuchos::null)
+          // we are on a rank that was dropped by removeEmptyProcesses
+          globalRank = globalSize;
+        // set the default output rank to the lowest of the available ranks
+        Teuchos::reduceAll<int,int>(*oldComm,Teuchos::REDUCE_MIN,globalRank,Teuchos::outArg(defaultMueLuOutputRank));
```

Comment:
It's kinda funny to use `Teuchos::reduceAll` in some places and `MPI_` functions in others.